src/App/pages/SingleRecipePage/SingleRecipePage.tsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import Loader from 'components/Loader';
+import FoodService from 'services/FoodService';
+
+// import { getTestRecipe } from 'utils/getTestRecipe';
+import { ISingleRecipe } from 'utils/types';
+
+import Characteristic from './components/Characteristic';
+import Directions from './components/Directions';
+import Divider from './components/Divider';
+import Header from './components/Header';
+import Summary from './components/Summary';
+import Supplies from './components/Supplies';
+
+import styles from './SingleRecipePage.module.scss';
+
+const SingleRecipePage = () => {
+  const [recipeData, setRecipeData] = useState<ISingleRecipe | null>(null);
+  const [isLoading, setLoading] = useState(true);
+
+  // ========================= Development with mock data =========================
+
+  // const foodService = new FoodService();
+
+  // useEffect(() => {
+  //   onRequest();
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, []);
+
+  // const onRequest = () => {
+  //   const rawRecipe = getTestRecipe();
+  //   const recipeData = foodService._transformSingleRecipeData(rawRecipe);
+
+  //   setRecipeData(recipeData);
+  //   setLoading(false);
+  // };
+
+  // ========================= Get real data from API =========================
+
+  const foodService = new FoodService();
+  const { id } = useParams();
+
+  useEffect(() => {
+    onRequest();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const onRequest = () => {
+    foodService.getRecipeById(id).then(onRecipeLoaded).catch(onError);
+  };
+
+  const onRecipeLoaded = (recipesData: ISingleRecipe) => {
+    setRecipeData(recipesData);
+    setLoading(false);
+  };
+
+  const onError = (error: Error) => {
+    // eslint-disable-next-line no-console
+    console.log(`Some error occured: ${error}`);
```

Comment:
лучше не инлайнить ошибку в строку, объект ошибки может нормально отобразиться
```suggestion
    console.log(`Some error occured:`, error);
```

Reply:
Исправил